Mathlib/Algebra/GroupWithZero/Int.lean

```diff
@@ -0,0 +1,33 @@
+/-
+Copyright (c) 2025 María Inés de Frutos-Fernández, Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández, Filippo A. E. Nuccio
+-/
+import Mathlib.Algebra.Group.Int.TypeTags
+import Mathlib.Algebra.GroupWithZero.WithZero
+
+/-!
+# Lemmas about `ℤₘ₀`.
+-/
+
+local notation "ℤₘ₀" => WithZero (Multiplicative ℤ)
+
+namespace WithZero
+
+open Multiplicative
+
+theorem ofAdd_zpow (a : ℤ) : (↑(ofAdd a) : ℤₘ₀) = ofAdd (1 : ℤ) ^ a := by
+  rw [← WithZero.coe_zpow, WithZero.coe_inj, ← Int.ofAdd_mul, one_mul]
+
+theorem ofAdd_zpow_zpow_comm (a b c : ℤ) :
+    ((↑(ofAdd a) : ℤₘ₀) ^ b) ^ c = (ofAdd (a : ℤ) ^ c) ^ b := by
+  simp only [← WithZero.coe_zpow]
+  rw [← zpow_mul, mul_comm, zpow_mul]
```

Comment:
I haven't opened this in Lean, but this looks to me like a general lemma about `GroupWithZero`s. Do you see what I mean, or am I mistaken?

Reply:
You are both right; I have generalized this lemma and moved it to `Algebra.GroupWithZero.WithZero`. Thanks!